lib/core/managementClient.ts

```diff
@@ -532,6 +556,209 @@ export class ManagementClient extends LinkEntity {
+            };
+            break;
+          case 83483426824:
+            rule.filter = {
+              expression: "1=0"
+            };
+            break;
+          case 83483426822:
+            if (Array.isArray(filtersRawData.value) && filtersRawData.value.length === 2) {
+              rule.filter = {
+                expression: filtersRawData.value[0] && filtersRawData.value[0].value
+              };
+            }
+            break;
+          case 83483426825:
+            if (Array.isArray(filtersRawData.value) && filtersRawData.value.length === 9) {
+              rule.filter = {
+                correlationId: filtersRawData.value[0] && filtersRawData.value[0].value,
+                messageId: filtersRawData.value[1] && filtersRawData.value[1].value,
+                to: filtersRawData.value[2] && filtersRawData.value[2].value,
+                replyTo: filtersRawData.value[3] && filtersRawData.value[3].value,
+                label: filtersRawData.value[4] && filtersRawData.value[4].value,
+                sessionId: filtersRawData.value[5] && filtersRawData.value[5].value,
+                replyToSessionId: filtersRawData.value[6] && filtersRawData.value[6].value,
+                contentType: filtersRawData.value[7] && filtersRawData.value[7].value,
+                userProperties: filtersRawData.value[8] && filtersRawData.value[8].value,
+              };
+            }
+            break;
+          default:
+            break;
+        }
+
+        if (actionsRawData.descriptor.value === 1335734829062
+          && Array.isArray(actionsRawData.value)
+          && actionsRawData.value.length) {
+          rule.action = {
+            expression: actionsRawData.value[0] && actionsRawData.value[0].value
+          };
+        }
+
+        rules.push(rule);
+
+      });
+
+      return rules;
+
+    } catch (err) {
+      const error = translate(err);
+      log.error("An error occurred while sending the get rules request to $management " +
+        "endpoint: %O", error);
+      throw error;
+    }
+  }
+
+  /**
+   * Removes the rule on the Subscription identified by the given rule name.
+   * @param ruleName
+   */
+  async removeRule(ruleName: string): Promise<void> {
+    try {
+      const request: AmqpMessage = {
+        body: {
+          "rule-name": types.wrap_string(ruleName)
+        },
+        reply_to: this.replyTo,
+        application_properties: {
+          operation: Constants.operations.removeRule
+        }
+      };
+      request.application_properties![Constants.trackingId] = generate_uuid();
+
+      log.mgmt("[%s] Remove Rule request body: %O.", this._context.namespace.connectionId,
+        request.body);
+      log.mgmt("[%s] Acquiring lock to get the management req res link.",
+        this._context.namespace.connectionId);
+      await defaultLock.acquire(this.managementLock, () => { return this._init(); });
+
+      await this._mgmtReqResLink!.sendRequest(request);
```

Comment:
Could this be turned into a boolean-returning function? if the request succeeds we can return true and return false otherwise?

Reply:
I had the same thought. Checked with Neeraj. The .NET SDK doesnt return anything either and expects the user to try/catch the call to find the failure case. 

There are other calls in this file which are similar and do not return anything. I decided to keep the same pattern for now and revisit when we have the API Review later

Reply:
I agree -- let's keep this consistent with the other SDKs.